pkg/envoy/agent.go

```diff
@@ -155,7 +155,13 @@ func (a *Agent) terminate() {
 				a.abortCh <- errAbort
 				return
 			}
-			log.Infof("There are still %d active connections", ac)
+			select {
+			case status := <-a.statusCh:
+				a.statusCh <- status
+				return
```

Comment:
To simplify this, can `terminate` always pull from `a.statusCh` and return the result, rather than making `Run` do it afterwards?
Also, it would be nice to pull from `a.statusChan` during the `terminationDrainDuration` sleep (which should then switch to a timer).

Reply:
I have no strong preference, done in https://github.com/istio/istio/pull/36691/commits/c8a6a428d4e77ea412ca7e4c738c37aac90738c6

Reply:
instead of `range ticker.C` should we select over `a.statusCh` and `ticker.C`? Otherwise we must always wait for a tick.